src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+
+    return visualization
+
+
+@dataclass
+class MarigoldDepthOutput(BaseOutput):
+    """
+    Output class for Marigold monocular depth prediction pipeline.
+
+    Args:
+        prediction (`PIL.Image.Image`, `np.ndarray`, `torch.FloatTensor`, `List[PIL.Image.Image]`, `List[np.ndarray]`,
+                or `List[torch.FloatTensor]`):
+            Predicted depth, with values in the range of [0, 1].
+        visualization (`None`, `PIL.Image.Image`, or List[PIL.Image.Image]):
+            Colorized prediction for visualization.
+        uncertainty (`None`, `np.ndarray`, torch.FloatTensor, or a `List` of them):
+            Uncertainty map computed from the ensemble.
+        latent (`None`, `torch.FloatTensor`, or `List[torch.FloatTensor]`):
+            Latent features corresponding to the ensemble predictions.
+        config (`None` or `dict`):
+            A set of configuration parameters used for this prediction. This can be useful for bookkeeping and
+            understanding the values resolved automatically, such as batch_size.
+    """
+
+    prediction: Union[
+        Image.Image,
+        np.ndarray,
+        torch.FloatTensor,
+        List[Image.Image],
+        List[np.ndarray],
+        List[torch.FloatTensor],
+    ]
+    visualization: Union[
+        None,
+        Image.Image,
+        List[Image.Image],
+    ]
+    uncertainty: Union[
+        None,
+        np.ndarray,
+        torch.FloatTensor,
+        List[np.ndarray],
+        List[torch.FloatTensor],
+    ]
+    latent: Union[
+        None,
+        torch.FloatTensor,
+        List[torch.FloatTensor],
+    ]
+    config: Dict[str, Any] = None
+
+
+class MarigoldDepthPipeline(DiffusionPipeline):
+    """
+    Pipeline for monocular depth estimation using the Marigold method: https://marigoldmonodepth.github.io.
+
+    This model inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods the
+    library implements for all the pipelines (such as downloading or saving, running on a particular device, etc.)
+
+    Args:
+        unet (`UNet2DConditionModel`):
+            Conditional U-Net to denoise the depth latent, conditioned on image latent.
+        vae (`AutoencoderKL`):
+            Variational Auto-Encoder (VAE) Model to encode and decode images and predictions to and from latent
+            representations.
+        scheduler (`DDIMScheduler`):
+            A scheduler to be used in combination with `unet` to denoise the encoded image latents.
+        text_encoder (`CLIPTextModel`):
+            Text-encoder, for empty text embedding.
+        tokenizer (`CLIPTokenizer`):
+            CLIP tokenizer.
+    """
+
+    def __init__(
+        self,
+        unet: UNet2DConditionModel,
+        vae: AutoencoderKL,
+        scheduler: DDIMScheduler,
+        text_encoder: CLIPTextModel,
+        tokenizer: CLIPTokenizer,
+    ):
+        super().__init__()
+
+        validate_vae(vae)
```

Comment:
When users will do `MarigoldPipeline.from_pretrained(...)`, they will already have the right VAE, no? I agree there's always a chance to manually set the `vae` attribute but we rely on the user to ensure they know what they are doing.

Reply:
Agreed!

Reply:
Fixed